Adding a set of guidelines to be kept in mind while organizing an event (#38)
kossiitkgp/docs: community/event-checklist.md

toc=true

+++

# 1. Logistics - Before the Workshop

> ✔️ Decide the **_date and time_** of the event as per the convenience of speakers and audience. Conducting **_events during exams_** or some other important event might **_not be a good idea_** in general. Checking **_academic calendar_** and other events in KGP is suggested.

Reviewer: Does markdown now also support emojis for list decoration or is it just a stylistic change? If so why not use the default unordered list decorators ex. `-` & `*`?

Author: The reason I used ✔️ for points was that I wanted the reader to feel them as a checklist rather than just some points. While conducting a workshop, keep this list in a dropbox paper or github issue and verify that all important things are checked.

It might seem trivial, but there are events in the past, where we have missed some of them. Hence I felt a check list was important.

Also, the checkmark emojis will be rendered on web too, when this doc will be a static site.

​      🗸 Make sure that something important that has been covered in the past doesn't gets removed.

​      🗸 Feel free to remove or edit stuff if you ever feel some part of the content is unnecessary and irrelevant, after discussing with team mates. 

✔️ Make sure to take feedback on the slides(content) from other members of KOSS. It is suggested to have the discussion over a meet to discuss things like potential queries from audience, thinking of ways to simplify content.

✔️ Refer [kossiitkgp/mistakes](https://github.com/kossiitkgp/mistakes) repository to get an idea of what sought of things can go wrong.

✔️ It is a good idea to record the event(offline/online) for the sake of posterity.

✔️ Make a Github repo for the workshop with name as event name and year which contains - installation instructions, slides, references, recording and others. (For example refer - [1](https://github.com/kossiitkgp/git-and-github-workshop-2020), [2](https://github.com/kossiitkgp/REST-APIs-in-Flask-Workshop-2021), [3](https://github.com/kossiitkgp/Linux-Shell-Vim-Workshop-2021))

> 

Reviewer 2: One thing which I used to do is that I would message the people who had shown interest in the topic being presented by asking me questions on it before the workshop was planned and oftentimes they did show up.

Author: The main aim of the repo was
- Audience can get all the information about the workshop like installation, slides and references at any time they want.
- Future KOSS members can know how the workshop was conducted and can have all the material for reference.

> message the people who had shown interest in the topic being presented

This is a nice idea. But messaging people will restrict. What was done for the recent workshops was that - we use to put the meeting link in the common group (of executives and CTMs) and interested people would join. And we would discuss slide by slide and see where potentially audience get confused and also each member presents his ideas to simplify things. 

This worked really well for the recent workshops - Linux, REST API workshop.  Hence I put this point in the doc, which covers it
![image](https://user-images.githubusercontent.com/44907757/117695744-2fef4c80-b1de-11eb-9bf8-e21a722b8e86.png)

- Meet regularly to discuss stuff. It is advisable to send mail invites even for a smaller meet, to avoid procrastination.

- **Making the workshop interactive**: It is always a good thing to keep the workshop interactive. If the workshop is conducted offline, we can have members of KOSS help people personally with hands on during the workshop and clear doubts more personally. But during an online workshop, this is not possible. So for sake of interaction, the speakers can encourage the audience to share the screen to show the work.  This acts as validation and also helps the rest of the audience. Asking the audience to share their screen in case of difficulties should also be encouraged. However note that,**do not force the audience to share their screen for sake of interaction. It has be voluntary and audience should not feel uncomfortable through such actions.**

# 2. Technicalities - Workshop content

## 2.1 How to decide the Topic for the workshop

- Make sure that the topic adds value to the community. The topic doesn't have to be trendy or in vogue.

> - Atleast a couple of members at KOSS should have enough practical experience to speak about the topic

Author: This is true, but the main thing I wanted to convey was that people who are presenting should have enough practical experience with the topic, no matter how suitable the topic is for the current needs of people.

An anecdote: In KOSS-18, we were planning to conduct a golang workshop, whilst we were writing KWoC backend in golang. But we dropped the idea, because none of us had experience of at least one project in golang. We felt, we were not enough experienced to teach to an audience.
Obviously, one can learn the basics and deliver the content. But that will not be effective. It will be as if, we were just reproducing the content. We might face trouble in answering the queries of the audience. In all our workshops, we use our experience and try to simplify topics we faced difficulties while learning(like Git - staging area, sending PR, branching, REST API).
Hence I felt the criteria that to deliver a workshop, practical experience is necessary.

> - Atleast a couple of members at KOSS should have enough practical experience to speak about the topic

Reviewer 2: @rakaar Practical experience with the topic before delivering a workshop is absolutely necessary.

> - Atleast a couple of members at KOSS should have enough practical experience to speak about the topic

Author: @pranitbauva1997 With the obsession of making this doc shorter, I didn't explain this point in detail(and at other places too). But sometimes, it is important to mention "why" when it is not straightforward, otherwise this doc would be like a set of doctrines. 
I added an explanation to this. If there are any other places, u find an explanation is required, like here it was, plz mention them.

## 2.2 How to decide the content of the workshop

> Make sure that the content adds a practical value to the audience. It might be tempting to add bits of cool jargon or theoretically detailed stuff to impress. But it is not recommended. For every slide, ask- Is this something that adds practical value. Will knowing or not knowing a particular thing affect the practical understanding of the topic. For example, consider explaining the idea of Merkel Tree in git, it might not be necessary to include the explanation of Merkel tree. As someone getting started with git, it is more important to practically understand staging area, branching, commits, sending a PR, pushing and pulling. When we get into details like Merkel Tree, audience might lose track of the event and lose interest. However, for the curious audience, it is a good idea to include such things at the end of the workshop, and provide further resources to explore more. 

Author: Exactly, it is important that we deliver what is important practically, rather than just delivering everything we know about the topic. I first heard about it in a podcast - https://www.scienceasylum.com/podcast/ATOTS001.mp3

"While the end of the semester, students just remember half of the content, and we should make sure that they remember the right half"

Since our time is limited, we should choose things of practical importance!

> During the core period of the workshop, make sure that they get a **practical hands on experience**(which is what they have come for, otherwise they could have learnt it on YouTube too!)

Reviewer 2: Doing this physically is easy because you can walk up to someone's desk and check on their progress and help them if they are stuck somewhere. But how is this achievable online?

Reviewer 2: @rakaar Good to know you have found a way to do this online as well.

toc=true

+++

# 1. Logistics - Before the Workshop

✔️Decide the date and time of the event as per the convenience of speakers and audience. Conducting events during exams or some other important event might not be a good idea in general. Checking academic calendar and other events in KGP is suggested.

✔️Publicize the event at least 2 or 3 days before the workshop. Publicizing earlier will fade away the event info from memory and publicizing late will be difficult for audience to plan.

✔️Share the Post via Facebook, Twitter and Linkedin via KOSS Account.

✔️Speakers should be decided atleast a week before the workshop, giving them enough time to prepare.

> ✔️It is important to have backup speakers for atleast 2 levels.(A backup speaker for the decided speaker. Another backup speaker for the backup speaker)

Reviewer 3: A backup for a backup seems like a stretch. Did anything such happen?

Author: Recently in a workshop, there was no backup speaker at all. And one of the members had to deliver an impromptu. It was done pretty well actually!

Having at least 2 backup speakers has another important side effect that there will be more people who are aware of the content of the workshop and hence can clarify doubts or can take up in emergency.

In some cases, like the python workshop/git workshop where it happens in 2 rooms, it is good if we have more people.

✔️Speakers of the event should have enough practical experience with the topic. The experience is necessary so that speaker can convey the content in the simplest way and clear doubts.

✔️It is highly recommended that the Slides for the event are made from scratch by the speaker.  Preparing from scratch will add creativity and newness to the content. New and simpler ways to present a topic might come out.  Also...

​      🗸Feel free to look up at previous slides at [kossiitkgp/slides](https://github.com/kossiitkgp/slides) repo for inspiration

​     🗸Make sure that something important that has been covered in the past doesn't gets removed.

​      🗸Feel free to remove or edit stuff if you ever feel some part of the content is unnecessary and irrelevant, after discussing with team mates. 

> ✔️Make sure to take feedback on the slides(content) from other members of KOSS regarding the slides. It is suggested to have the discussion over a meet to discuss things like potential queries from audience, thinking of ways to simplify content.

Reviewer 4: ```suggestion
✔️Make sure to take feedback on the slides(content) from other members of KOSS regarding the slides. It is suggested to have the discussion over a **_Google Meet(or any other medium)_** to discuss things like potential queries from the audience, thinking of ways to simplify content and so on.
```

✔️Refer [kossiitkgp/mistakes](https://github.com/kossiitkgp/mistakes) repository to get an idea of what sought of things can go wrong.

✔️It is a good idea to record the event(offline/online) for the sake of posterity.

✔️Make a Github repo for the workshop with name as event name and year which contains - installation instructions, slides, references, recording and others. (For example refer - [1](https://github.com/kossiitkgp/git-and-github-workshop-2020), [2](https://github.com/kossiitkgp/REST-APIs-in-Flask-Workshop-2021), [3](https://github.com/kossiitkgp/Linux-Shell-Vim-Workshop-2021))

## 1.1 Internal

> - Keep in touch regularly with other team mates on updates regarding the event. Regular communication on #events channel is suggested.

Reviewer 4: ```suggestion
- Keep in touch regularly with other team mates on updates regarding the event. Regular communication on **#events** channel is suggested.
```

> - Meet regularly to discuss stuff. It is advisable to send mail invites even for a smaller meet, to avoid procrastination.

Reviewer 4: ```suggestion
- Meet regularly to discuss stuff. It is advisable to send **_calendar invites_** even for a smaller meet, to avoid procrastination.
```

# 2. Technicalities - Workshop content

## 2.1 How to decide the Topic for the workshop

> - Make sure that the topic adds value to the community. The topic doesn't have to be sexy.

Reviewer 3: Can we use a word more apt then 'sexy'? Maybe 'trendy' or 'in vogue'?  
> The topic doesn't have to be trendy or clickbaity.

Author: Replaced with the words u had suggested! Thanks!
U are right, "Sexy" is a vague word

> - **Making the workshop interactive**: It is always a good thing to keep the workshop interactive. If the workshop is conducted offline, we can have members of KOSS help people personally with hands on during the workshop and clear doubts more personally. But during an online workshop, this is not possible. So for sake of interaction, the speakers can encourage the audience to share the screen to show the work, for kind of validation and it also helps the rest of the audience. Asking the audience to share their screen in case of difficulties should also be encouraged. However note that,**do not force the audience to share their screen for sake of interaction. It has be voluntary and audience should not feel uncomfortable through such actions.**

Reviewer 4: ```suggestion
- **Making the workshop interactive**: It is always a good thing to keep the workshop interactive. If the workshop is conducted offline, we can have members of KOSS help people personally with hands on during the workshop and clear doubts more personally. But during an online workshop, this is not possible. So for sake of interaction, the speakers can **_encourage the audience_** to **_share the screen_** to show the work, for kind of validation and it also helps the rest of the audience. Asking the audience to share their screen in case of difficulties should also be encouraged. However note that,**do not force the audience to share their screen for sake of interaction. It has be _voluntary_ and audience should _not feel uncomfortable_ through such actions.**
```

> - Atleast a couple of members at KOSS should have enough practical experience to speak about the topic

Reviewer 4: ```suggestion
- Atleast a couple of members at KOSS should have enough practical experience to speak about the topic.
```

## 2.2 How to decide the content of the workshop

> Make sure that the content adds a practical value to the audience. It might be tempting to add bit of cool jargon or theoretically detailed stuff to impress. But it is not suggested. For every slide, ask - Is this something that adds practical value. Will knowing or not knowing a particular thing affect the practical understanding of the topic. For example, consider explaining the idea of Merkel Tree in git, it might not be necessary to include the explanation of Merkel tree. As someone getting started with git, it is more important to practically understand staging area, branching, commits, sending a PR, pushing and pulling. When we get into details like Merkel Tree, audience might loose track of the event and loose interest. However, for the curious audience, it is a good idea to include such things at the end of the workshop, and provide further resources to explore more. 

Reviewer 4: ```suggestion
Make sure that the content adds a practical value to the audience. It might be tempting to add bit of cool jargon or theoretically detailed stuff to impress. **_But it is not suggested_**. 
For every slide, ask - Is this something that adds practical value. Will knowing or not knowing a particular thing affect the practical understanding of the topic. **_For example, consider explaining the idea of Merkel Tree in git, it might not be necessary to include the explanation of Merkel tree._** As someone getting started with git, it is more important to practically understand staging area, branching, commits, sending a PR, pushing, and pulling. When we get into details like Merkel Tree, the audience might lose track of the event and lose interest. However, for the curious audience, it is a good idea to include such things at the end of the workshop, and provide further resources to explore more. 
Make sure the content is as simple as it can be, and at the end, small enough such that it could be summarised easily :)
```

During the core period of the workshop, make sure that they get a **practical hands on experience**(which is what they have come for, otherwise they could have learnt it on YouTube too!)

Make sure to give enough time to cover the queries and concerns of the audience during the workshop.(without queries and interaction, the workshop would not differ from a YouTube Tutorial)

## 2.3 How to organize or approach content:

> - **2.3.1 Introduction:** It is a good idea to tell the audience the following things: 

Reviewer 4: ```suggestion
- **2.3.1 Introduction:** It is a good idea to tell the audience the following things before hand: 
```

>   - What they should take away from the workshop. For example, if it is git workshop, convey that by the end of the workshop they will be comfortable to track their project using git and know how to send a Pull Request properly to a open source repository on Github.

Reviewer 4: ```suggestion
  - What should they expect from the workshop. For example, if it is **_Git_** workshop, convey that by the end of the workshop they will be comfortable to track their project using Git and know how to send a Pull Request properly to a open-source repository on GitHub.
```

>   - Give the audience about how the flow of workshop would be. For example...

Reviewer 4: ```suggestion
  - Give the audience about how the flow of workshop would be. For example - 
```
don't use ellipsis, although indians are more prone to use them... [link](https://www.reddit.com/r/indianpeoplefacebook/comments/5j0bn4/why_do_indians_overuse_ellipses_in_their_writing/)

>     - While hands on, asking them to do them along with the speaker

Reviewer 4: ```suggestion
    - While hands on, asking them to do them along with the speaker.
```

    - There will be small doubt sessions periodically **or** may be if the speaker is comfortable, he/she may ask the audience to interrupt in middle and ask the query. It is a good idea to take breaks after each [microcentury](https://susam.in/blog/microcentury/)
    - It is a good idea to warn the audience if there is a long strech of theoretical discussion for the next few minutes.

>   - **2.3.2 Introducing a new topic**: If you are introducing any new topic, it is suggested to try either of the two methods

Reviewer 4: ```suggestion
  - **2.3.2 Introducing a new topic**: If you are introducing any new topic, it is suggested to try either of the two methods - 
```

>     - **First principles approach**: Introducing the topic from the ground up. Don't just bring up a word and say, "we are going to learn about this X topic now". Bring the situation to explain why something of this sought is necessary. For example, in git consider the topic "branching". Introducing a problem in the beginning, where git branching is a super-useful solution, this will help the audience grasp the idea better.

Reviewer 4: ```suggestion
    - **First principles approach**: Introducing the topic from the ground up. Don't just bring up a word and say, **_"we are going to learn about this X topic now"_**. Bring the situation to explain why something of this sought is necessary. For example, in git consider the topic "branching". Introducing a problem in the beginning, where git branching is a super-useful solution, this will help the audience grasp the idea better.
```
Love this ❤️

    - **Using Analogy**: It is always a useful thing to give a analogy with something which people already know. For  example, in git consider the idea of staging area. The staging area is [generally difficult to grasp](https://github.com/kossiitkgp/mistakes/blob/master/git-workshop.md#git-workshop-2018) for someone new to git. To explain this, one can use "shopping in a mall"  analogy. 

      - Selecting items from the rack <-> Making changes in the files

      - Adding items in the shopping cart <-> Adding files in the staging area using `git add `

      - Billing at the counter <-> Commiting the changes

>         The above analogy was inspired from a [stack exchange answer by Spoike](https://softwareengineering.stackexchange.com/questions/119782/what-does-stage-mean-in-git), Later modified by Pankaj Kumar. Good analogies can be found by searching for intuitive explanations for the topic on web and also by brainstorming with the team. The above analogy was a result of both!

Reviewer 4: ```suggestion
        The above analogy was inspired from a [stack exchange answer by Spoike](https://softwareengineering.stackexchange.com/questions/119782/what-does-stage-mean-in-git), Later modified by @Shankusu993 . Good analogies can be found by searching for intuitive explanations for the topic on web and also by brainstorming with the team. The above analogy was a result of both!
```

Author: I was afraid that his username might change. But his original name will never change! Hence I put that
But ya, putting by username is a cool thing

If @Shankusu993 , if u plan to change ur username in future. U can add ur name. 
Currently, accepting the commit suggestion - using username

        **Caution**: **Make sure to be aware of the consequences of the analogy used. Sometimes analogies can be misguiding. In that case, it is suggested to chose a different analogy. Alternatively, you can also warn the audience that the crafted analogy does not hold for further ideas. The analogy is only helpful for this concept.**

    # 3.  Logistics - After workshop

>     ✔️Upload the slides and reference in the Github repo made for the workshop

Reviewer 4: ```suggestion
    ✔️Upload the slides and reference in the GitHub repo made for the workshop.
```

>     ✔️Once audience has learnt the basics from our workshop. We should encourage the curious to explore more advanced and interesting topics on their own. Hence for such people, update further reading material or links to explore more regarding the topic.

Reviewer 4: ```suggestion
    ✔️Once the audience has learned the basics from our workshop, we should encourage the curious ones to explore more advanced and interesting topics on their own. Hence for such people, update further reading material or links to explore more regarding the topic.
```
